### ISSUE TITLE
Feasibility of alt wps pin attack

### ISSUE THREAD

**Issue by ghost** [soxrok2212/PSKracker#5]
hello, I saw you your work in wps pentesting, so i would like to know from you if what is described here is feasible or if it has already been tried before ( i am aware only of the wps online bruteforce and pixie dust )
[url]https://pdfs.semanticscholar.org/14db/881399d414e4ea44db7e97d672241174c999.pdf[/url]  
Its called "New attacks on Wi-Fi Protected Setup" if you can't access with that link.

**Comment by soxrok2212** [comment]
This paper was not written by very professional researchers. On page 131, the author writes "But if the attacker could somehow change his role in this exchange and could be the enrollee, he can use two arbitrary values instead of E-Hash1 and E- Hash2 and sends it as M3 message." They don't even propose an idea as to how the attacker could "somehow" take role as the enrollee. Look at the protocol and you will see that you can't just switch roles, and the Wi-Fi Alliance thought of this beforehand. The common misconception is that the AP is the enrollee and the client is the registrar (I guess the naming scheme seems a bit backwards). This was actually proposed at the WPS Blackjack Attack some years ago but the link no longer works and was debunked very quickly after being publicized. 

TL;DR no. You can not just switch roles, and no, something like what that PDF suggests is not possible.
